tests/unit/util/util_test.py

```diff
@@ -415,6 +425,105 @@ def test_refill(self):
         self.advance_time(1)
         self.assertTrue(self.charge_if_available(60))
 
+    def test_basic_use_sleep(self):
+        def fake_sleep(self2, seconds):
```

Comment:
Using mock patch, there's an easier way to do this now.  Search for `@mock.patch("scalyr_agent.scalyr_client.time.time", mock.Mock(return_value=0))` in the unittests to see examples of how you can override methods with decorators and context objects.  Since you actually need to advance time, you probably should use a context manager.  It would look something like this:

```
  with mock.patch("scalyr_agent.util.time.sleep", self.fake_sleep):
      ## Do the rest of your test here.
``` 

In this example, I'm assuming you define a `fake_sleep` method at the `TestRateLimiter` level.  I'm not quite sure why you keep declaring `fake_sleep` within the method.. I think you just need to do it once and reuse that.

The context object will handle cleaning up the patch state.. so you don't have to do the `try .. finally` yourself.

We could also use this approach to patch `time.time` as well so that we don't have to pass in `current_time` everywhere.  I would mark that as a TODO for now... I'm not crazy about using different testing approaches in the same class (i.e., passing down current time to override time.time vs patching time.sleep) but it's going to be a little bit of work to change all of this.

Reply:
The redefinition was due to some weirdness with how many parameters the method ends up expecting when defined outside the method, probably won't be an issue with this approach.